tests/ipv6/run-ipv6-test.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+set -e # Exit immediately if a command exits with a non-zero status
+
+RELEASE=$1
+RELEASE_IMAGE=$2
+
+if [[ -z $RELEASE || -z $RELEASE_IMAGE ]]; then
+    echo "Usage: $0 <RELEASE> <RELEASE_IMAGE> [HELM_RELEASE_NAME] [NAMESPACE] [CLUSTER_NAME]"
+    echo "Error: RELEASE and RELEASE_IMAGE are required parameters. Example usage $(make ipv6-test RELEASE=vX.Y.Z RELEASE_IMAGE=release-X.Y-rc)"
+    exit 1
+fi
+
+HELM_RELEASE_NAME=${3:-ngf}
+NAMESPACE=${4:-nginx-gateway}
+CLUSTER_NAME=${5:-ipv6-only-${RELEASE}}
+RELEASE_REPO=ghcr.io/nginx/nginx-gateway-fabric
+
+cleanup() {
+    echo "Cleaning up resources..."
+    kubectl delete -f ipv6/manifests/ipv6-test-app.yaml || true
+    kubectl delete -f ipv6/manifests/ipv6-test-client.yaml || true
+    kubectl delete -f ipv6/manifests/gateway.yaml || true
+    helm uninstall ${HELM_RELEASE_NAME} -n ${NAMESPACE} || true
+    kind delete cluster --name ${CLUSTER_NAME} || true
+}
+
+trap cleanup EXIT
+
+kind create cluster --name ${CLUSTER_NAME} --config ipv6/config/kind-ipv6-only.yaml
+
+echo "Applying Gateway API CRDs"
+kubectl kustomize "https://github.com/nginx/nginx-gateway-fabric/config/crd/gateway-api/standard?ref=${RELEASE}" | kubectl apply -f -
+
+docker pull ${RELEASE_REPO}:${RELEASE_IMAGE}
+
+docker save ${RELEASE_REPO}:${RELEASE_IMAGE} | docker exec -i ${CLUSTER_NAME}-control-plane ctr --namespace=k8s.io images import -
+
+helm upgrade --install ${HELM_RELEASE_NAME} oci://ghcr.io/nginx/charts/nginx-gateway-fabric \
+    --create-namespace -n ${NAMESPACE} \
+    --set nginx.config.ipFamily=ipv6 \
+    --set nginx.service.type=ClusterIP \
+    --set nginxGateway.image.repository=${RELEASE_REPO} \
+    --set nginxGateway.image.tag=${RELEASE_IMAGE}
+
+echo "Deploying Gateway..."
+kubectl apply -f ipv6/manifests/gateway.yaml
+
+echo "Waiting for NGINX Gateway to be ready..."
+kubectl wait --for=condition=accepted --timeout=300s gateway/gateway
+POD_NAME=$(kubectl get pods -l app.kubernetes.io/instance=${HELM_RELEASE_NAME} -o jsonpath='{.items[0].metadata.name}')
+kubectl wait --for=condition=ready --timeout=300s pod/${POD_NAME}
+
+echo "Deploying IPv6 test application"
+kubectl apply -f ipv6/manifests/ipv6-test-app.yaml
+
+echo "Waiting for NGF to be ready..."
+kubectl wait --for=condition=available --timeout=300s deployment/${HELM_RELEASE_NAME}-nginx-gateway-fabric -n ${NAMESPACE}
+
+echo "Waiting for test applications to be ready..."
+kubectl wait --for=condition=available --timeout=300s deployment/test-app-ipv6
+
+echo "Deploying IPv6 test client"
+kubectl apply -f ipv6/manifests/ipv6-test-client.yaml
+kubectl wait --for=condition=ready --timeout=300s pod/ipv6-test-client
+
+echo "Getting NGF service IPv6 address"
+NGF_IPV6=$(kubectl get service gateway-nginx -o jsonpath='{.spec.clusterIP}')
```

Comment:
This works, though we could enhance this test a little bit by getting the address from the Gateway status instead.

Reply:
Ah okay, that's good to know.
I'll try that out too!